modmesh/app/euler1d.py

```diff
@@ -29,101 +29,260 @@
 __package__ = "modmesh.app"
 
 import sys
-from dataclasses import dataclass
-
-import numpy as np
-
 import matplotlib
 import matplotlib.pyplot
+import numpy as np
+import modmesh as mm
+
+from dataclasses import dataclass
 from matplotlib.backends.backend_qtagg import FigureCanvas
-from matplotlib.ticker import FormatStrFormatter
 from matplotlib.figure import Figure
-from PySide6.QtWidgets import (QApplication, QWidget, QScrollArea,
-                               QGridLayout, QVBoxLayout, QPushButton,
-                               QHBoxLayout, QFileDialog, QLayout, QCheckBox)
-from PySide6.QtCore import Qt, QEvent, QMimeData, QTimer, QObject, Slot
-from PySide6.QtGui import QDrag, QPixmap
-
-import modmesh as mm
-from .. import view
+from matplotlib.ticker import FormatStrFormatter
+from PySide6.QtCore import QTimer, Slot, Qt
+from PySide6.QtGui import QPixmap
+from PySide6.QtWidgets import QFileDialog, QDockWidget
+from PUI.state import State
+from PUI.PySide6.base import PuiInQt, QtInPui
+from PUI.PySide6.button import Button
+from PUI.PySide6.layout import VBox, Spacer
+from PUI.PySide6.scroll import Scroll
+from PUI.PySide6.window import Window
+from PUI.PySide6.combobox import ComboBox, ComboBoxItem
+from PUI.PySide6.label import Label
+from PUI.PySide6.table import Table
+from PUI.PySide6.toolbar import ToolBar, ToolBarAction
 from ..onedim import euler1d
-
-
-def load_app():
-    view.mgr.pycon.writeToHistory("""
-# Use the functions for more examples:
-ctrl.start()  # Start the movie
-ctrl.step()  # Stepping the solution
-
-# Note: you can enable profiling info by "profiling" option
-ctrl = mm.app.euler1d.run(interval=10, max_steps=50, profiling=True)
-""")
-    cmd = "ctrl = mm.app.euler1d.run(interval=10, max_steps=50)"
-    view.mgr.pycon.command = cmd
+from .. import view
 
 
 @dataclass
 class QuantityLine:
+    """
+    A class representing a quantity line with associated data.
+
+    This class is a data structure that holds information about a quantity line
+    in a plot. It is designed to be used in conjunction with Matplotlib for
+    visualizing analytical and numerical data.
+
+    Class attributes:
+        - `ana` (matplotlib.lines.Line2D): Line2D for analytical data.
+        - `num` (matplotlib.lines.Line2D): Line2D for numerical data.
+        - `axis` (matplotlib.pyplot.axis): Axis for the plot.
+        - `name` (str): Name of the quantity.
+        - `unit` (str): Unit of measurement.
+
+    Methods:
+        - :meth:`update(xdata, adata, ndata)`: Update the line data and
+          redraw the plot.
+    """
     ana: matplotlib.lines.Line2D = None
     num: matplotlib.lines.Line2D = None
     axis: matplotlib.pyplot.axis = None
     name: str = ""
     unit: str = ""
 
-    def bind_axis(self, axis):
-        self.axis = axis
-
```

Comment:
Removing bind_axis method, assign axis directly in Line [381](https://github.com/solvcon/modmesh/pull/271/files#diff-4ebeb3148c683103bf030637fdf520154a4b12d9b660a82610861aa5250a0be1R381) and [442](https://github.com/solvcon/modmesh/pull/271/files#diff-4ebeb3148c683103bf030637fdf520154a4b12d9b660a82610861aa5250a0be1R442)
Just like assign `ana` and `num`.